nbconvert/filters/markdown_mistune.py

```diff
@@ -21,7 +21,42 @@
 from nbconvert.filters.strings import add_anchor
 
 
+class MathBlockGrammar(mistune.BlockGrammar):
+    """This defines a single regex comprised of the different patterns that 
+    identify math content spanning multiple lines. These are used by the 
+    MathBlockLexer.
+    """
+    multi_math_str = "|".join([r"^\$\$.*?\$\$",
+                               r"^\\\\\[.*?\\\\\]",
+                               r"^\\begin\{([a-z]*\*?)\}(.*?)\\end\{\1\}"])
+    multiline_math = re.compile(multi_math_str, re.DOTALL)
+
+
+class MathBlockLexer(mistune.BlockLexer):
+    """ This acts as a pass-through to the MathInlineLexer. It is needed in 
+    order to avoid other block level rules splitting math sections apart. 
+    """
+    
+    default_rules = (['multiline_math']
+                     + mistune.BlockLexer.default_rules)
+
+    def __init__(self, rules=None, **kwargs):
+        if rules is None:
+            rules = MathBlockGrammar()
+        super(MathBlockLexer, self).__init__(rules, **kwargs)
+
+    def parse_multiline_math(self, m):
+        """Add token to pass through mutiline math."""
+        self.tokens.append({
+            "type": "multiline_math",
+            "text": m.string
```

Comment:
I think this should be `m.group(0)`, which is the full match, not `m.string`, which is the string it tried to match against.

```
 3│ re.match(r'\d+-', '12-AB').string
 3> '12-AB'

 4│ re.match(r'\d+-', '12-AB').group(0)
 4> '12-'
```

I haven't checked how mistune works, but `m.string` could be wrong, whereas `m.group(0)` can't be. So let's go with the more robust option.